File: libs/libmpsse/src/examples/gpio.c
#include "mpsse_config.h"
#include <stdio.h>
#include <stdlib.h>
#include <mpsse.h>

int main(void)
{
	struct mpsse_context *io = NULL;
	int i = 0, retval = EXIT_FAILURE;

	io = MPSSE(GPIO, 0, 0);
	
	if(io && io->open)
	{
		for(i=0; i<10; i++)
		{
			PinHigh(io, GPIOL0);
			printf("GPIOL0 State: %d\n", PinState(io, GPIOL0, -1));
			sleep(1);
			
			PinLow(io, GPIOL0);
			printf("GPIOL0 State: %d\n", PinState(io, GPIOL0, -1));
			sleep(1);
		}
	
		retval = EXIT_SUCCESS;
	}
	else
	{
		printf("Failed to open MPSSE: %s\n", ErrorString(io));
	}
		
	Close(io);

	return retval;
}

File: libs/libmpsse/src/examples/gpio.py
#!/usr/bin/env python2

from mpsse import *
from time import sleep

# Open mpsse in GPIO mode
io = MPSSE(GPIO)

# Toggle the first GPIO pin on/off 10 times
for i in range(0, 10):

    io.PinHigh(GPIOL0)
    print "GPIOL0 State:", io.PinState(GPIOL0)
    sleep(1)

    io.PinLow(GPIOL0)
    print "GPIOL0 State:", io.PinState(GPIOL0)
    sleep(1)

io.Close()
